standards/gitcoinGrants948.md

```diff
@@ -0,0 +1,127 @@
+# [WIP] ERC948 Subscription Standard (v1)
+
+## Summary
+The following is a proposed standard for subscription payments with read and write specs.
+
+## Abstract
+This standard allows a user to set a recurring payment to a service provider. This recurring transaction is ideally executed by a third party agent, but at a minimum the service provider themselves can execute the payment. Either side of the subscription should be able cancel the agreement with a reasonable amount of forewarning to the other party. This standard or similar should be integrated with future smart wallets to enable subscription payments in the Ethereum ecosystem.
+
+## Motivation
+Recurring payments are a widely use monetization method in traditional software. Currently, no such pattern exists with desirable UX traits in the decentralized space. Creating such a pattern will unlock many new avenues for revenue and provide an alternative to the dominant rent seeking and token-valuation based monetization methods in the current ecosystem.
+
+### ERC948 Read
+
+```
+interface ERC165 {
+
+@notice Query if a contract implements an interface
+@param interfaceID The interface identifier, as specified in ERC-165
+@dev Interface identification is specified in ERC-165. This function uses less than 30,000 gas.
+@return `true` if the contract implements `interfaceID` and `interfaceID` is not 0xffffffff, `false` otherwise
+
+function supportsInterface(bytes4 interfaceID) external view returns (bool);
+
+}
+
+interface ERC948Read is ERC165 {
+
+@dev Return the details of a subscription
+@param subscriptionId - A unique representation of the subscription based on the details of the agreement
+
+function getSubscription(
+   bytes subscriptionId
```

Comment:
You probably meant `bytes32`. Var length arrays can only be passed in internal function calls. I think it would be a good idea to consider switching to `uint256` (see rationale: https://github.com/EthereumOpenSubscriptions/standard/blob/99d833792ddd4d2a422fea66fb21379c7a2ef61a/standards/stef_and_luka-standard_proposal_draft.md#subscription-identifiers)

Reply:
I did mean bytes32. If sh3 hashes are directly convertible to unit256 then it makes sense to use them. Can you point me to resources to learn more about this?

Reply:
It's a design decision, EVM works with 32 byte words (256bits)[1]. Since `bytes32` and `uint256` both use 2^256 bits conversion is free[2].  SO question about conversion: https://ethereum.stackexchange.com/questions/6498/how-to-convert-a-uint256-type-integer-into-a-bytes32

1. https://ethereum.stackexchange.com/a/3071/31630
2. https://ethereum.stackexchange.com/questions/48263/does-it-cost-more-gas-to-use-operators-with-bytes32-than-it-does-with-u#comment57178_48263